program/rust/src/tests/test_sma.rs

```diff
@@ -0,0 +1,168 @@
+use quickcheck::Arbitrary;
```

Comment:
🥇 for this generative test. This gives me a lot of confidence in the implementation.